Feature/autocomplete (#20)
MissouriMRR/propguard: README.md

[![Stargazers][stars-shield]][stars-url]
[![Issues][issues-shield]][issues-url]

<!-- PROJECT LOGO AND TITILE -->
<!-- TODO: Add project logo and title-->
<br />
<p align="center">
    <h2 align="center">Propguard</h2>
    <p align="center">Currently only two software members really know how to start the drone and enable its autonomous features. We need to change that. Propguard is a webapp that teaches people how to use Dronekit and Python to start and launch the drones.</p>
</p>

## Table of Contents

- [What is this?](#what-is-this)
- [Usage](#usage)
- [Get Started](#get-started)
- [Roadmap and Progress](#roadmap-and-progress)
- [Contributing](#contributing)

## What is this?

> Propguard is a webapp built for Missouri S&T's Multirotor Drone Design Team meant to preserve the knowledge and experience of its past members. This means that Propguard teaches new members how to start working on the team's custom flight code via a Codecademy-like tutorial interface.

Reviewer: ```suggestion
Propguard is a web app built for Missouri S&T's Multirotor Robot Design Team meant to preserve the knowledge and experience of its past members. Propguard teaches new members how to start working on the team's flight code via a Codeacademy-like tutorial interface.
```

> A second part of Propguard (in development) is to create a wiki that will serve as the collective knowledge of all the past members of the Multirotor team. During our flight tests we always run into issues, and some of those issues happen to be things that have been solved in the past. We obviously don't want to keep making the same mistakes.

Reviewer: ```suggestion
It includes a wiki that will serve as the collective knowledge of all the past members of the team. During our flight tests we always run into issues, and some of those issues happen to be things that have been solved in the past. We obviously don't want to keep making the same mistakes.
```

The tech stack for the front-end is Typescript and ReactJS.

## Usage

We're working on getting a live version up so for now you'll have to run this project locally on your machine. See the **Get Started** section for instructions on how to run it.

## Get started

### Requirements

- NodeJS
  - https://nodejs.org/en/
- Git
  - https://git-scm.com/book/en/v2/Getting-Started-Installing-Git
- Yarn
  - https://yarnpkg.com/lang/en/docs/install/

### Installation

1. Clone the repository to a folder of your choosing

```sh
git clone https://github.com/MissouriMRR/propguard.git
```

2. Install the required packages for the project

```sh

> yarn

Reviewer: Be explicit when writing instructions, this works but explicit is better
```suggestion
yarn install
```

```

3. Run the development version of Propguard

```sh
yarn start
```

4. Now play around, code, and enjoy!

[stars-shield]: https://img.shields.io/github/stars/MissouriMRR/propguard
[stars-url]: https://github.com/MissouriMRR/propguard/stargazers
[issues-shield]: https://img.shields.io/github/issues/MissouriMRR/propguard
[issues-url]: https://github.com/othneildrew/Best-README-Template/issues

## Roadmap and progress

> ### Winter Break Goals

Reviewer: Give dates

- Finish front-end for the tutorial part of Propguard

> ### Before Spring Break

Reviewer: Give dates

- Write basic tutorials for setting up development environments for the flight team
- Create the front-end for the wiki

> ### Before Finals

Reviewer: Give dates

- Write a tutorial on writing actual flight code
- Create a backend for the wiki to host content on
- Come up with a system that lets the wiki by easily editable with some CMS like Ghost+Gatsby or something similar

> ### Summer 2020 Before Competition

Reviewer: Give dates

[![Stargazers][stars-shield]][stars-url]
[![Issues][issues-shield]][issues-url]

<!-- PROJECT LOGO AND TITILE -->
<br />
<p align="center">
    <h2 align="center">Propguard</h2>

>     <p align="center">Currently only two software members really know how to start the drone and enable its autonomous features. We need to change that. Propguard is a webapp that teaches people how to use Dronekit and Python to start and launch the drones.</p>

Reviewer: On reevaluation, lets ditch the "two software members" and just keep a version of the last sentence. It's too specific and won't be true all the time.

</p>

## Table of Contents

- [What is this?](#what-is-this)
- [Usage](#usage)
- [Get Started](#get-started)
- [Roadmap and Progress](#roadmap-and-progress)
- [Contributing](#contributing)

## What is this?

Propguard is a web app built for Missouri S&T's Multirotor Robot Design Team meant to preserve the knowledge and experience of its past members. Propguard teaches new members how to start working on the team's flight code via a Codeacademy-like tutorial interface.

It includes a wiki that will serve as the collective knowledge of all the past members of the team. During our flight tests we always run into issues, and some of those issues happen to be things that have been solved in the past. We obviously don't want to keep making the same mistakes.

The tech stack for the front-end is Typescript and ReactJS.

## Usage

> We're working on getting a live version up so for now you'll have to run this project locally on your machine. See the **Get Started** section for instructions on how to run it.

Reviewer: You use use header links to link down to get started

## Get started

### Requirements

- NodeJS
  - https://nodejs.org/en/
- Git
  - https://git-scm.com/book/en/v2/Getting-Started-Installing-Git
- Yarn
  - https://yarnpkg.com/lang/en/docs/install/

### Installation

1. Clone the repository to a folder of your choosing

```sh
git clone https://github.com/MissouriMRR/propguard.git
```

2. Install the required packages for the project

```sh
yarn install
```

3. Run the development version of Propguard

```sh
yarn start
```

4. Now play around, code, and enjoy!

[stars-shield]: https://img.shields.io/github/stars/MissouriMRR/propguard
[stars-url]: https://github.com/MissouriMRR/propguard/stargazers
[issues-shield]: https://img.shields.io/github/issues/MissouriMRR/propguard
[issues-url]: https://github.com/othneildrew/Best-README-Template/issues

## Roadmap and progress

- Finish front-end for the tutorial part of Propguard (Feb 11)
- Write basic tutorials for setting up development environments for the flight team (Feb 25)
- Write a tutorial for writing actual flight code (March 10)
- Create a backend and database to host propguard tutorial (March 17)
- Find a place to host the wiki and tutorial (March 19)

> - Come up with a system that lets the wiki by easily editable with some CMS like Ghost+Gatsby or something similar (April 14)

Reviewer: Is this still accurate

Reviewer: Not sure that the roadmap should be in the repo anyways, you could do it by using projects and issues in github

<br />
<p align="center">
    <h2 align="center">Propguard</h2>

>     <p align="center">Propguard is a webapp that teaches people how to start writing basic flight code with technologies like Dronekit, MAVSDK, and Python.</p>

Reviewer: ```suggestion
    <p align="center">Propguard is a webapp that teaches people how to start writing basic flight code with technologies like PX4, MAVSDK, and Python.</p>
```
